### PR TITLE
Improve build-info comparison

### PR DESCRIPTION
**Pull request by Or-Geva**
- [x] All [tests](https://github.com/jfrog/build-info-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: utils/compare/compare.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package compare
```
Comment (eyalbe4) [inline utils/compare/compare.go:1]: I'm not sure we really need a new package for a single file.

File: utils/compare/compare.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import (
+	"reflect"
+	"sort"
+)
+
+// Check for equality on slices without order.
+func IsEqualSlices(a, b []string) bool {
```
Comment (eyalbe4) [inline utils/compare/compare.go:9]: Let's add a unit test for this function.

File: utils/compare/compare.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	copy(b_copy, b)
+
+	sort.Strings(a_copy)
+	sort.Strings(b_copy)
+
+	return reflect.DeepEqual(a_copy, b_copy)
+}
+
+func IsEqual2DSlices(a, b [][]string) bool {
+	return IsEqualSlices(to1DSlice(a), to1DSlice(b))
```
Comment (eyalbe4) [inline utils/compare/compare.go:27]: Let's add a unit test for this function.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: utils/compare/compare.go (hunk 1 above)
Comment (Or-Geva) [reply utils/compare/compare.go]: Ok, but I had to move npm utils to the build utils because otherwise, we will have an import cycle.

Comment (eyalbe4) [reply utils/compare/compare.go]: Got it. In that case, let's leave it as is,

**Review by eyalbe4 — APPROVED** [review]

**Review by eyalbe4 — APPROVED** [review]

**Pull request merged**